### ISSUE TITLE
Ctrl + left click not working with sticky keys

### ISSUE THREAD

**Issue by lytex** [kmonad/kmonad#542]
Basically what the title says, if I configure a key to be `rctl` everything works fine, but when defined as `(sticky-key 500 rctl)` for example, ctrl+left click no longer works

**Comment by david-janssen** [comment]
Yeah, that makes sense. We only ever check keyboard input, and sticky keys are designed to do nothing unless followed some event. Since we never catch mice-event, a `sticky-key -> mouse-event` just looks like a sticky-key without follow-up to KMonad.

I never used sticky keys myself, so I have a question.

Is it important that the key *do nothing* if not followed by another key? I.e. if we take your `(sticky-key 500 rctl)` and we were to tap it, there are 2 ways of getting a 'sticky effect`. The current way is that we wait for an event to happen and then wrap it in `rctl`. However, since we can't see mice events we can't wrap them.

The other way of doing this is making a key that, when tapped, simple presses `rctl` and then releases it 500 ms later. That way the stickiness won't go away immediately after the event (it will always persist for 500 ms), but we wouldn't have to catch an event to trigger the mod-state, so if you press nothing your system would still be in `rctl`-is-pressed for 500 ms, and if you click your mouse in that window it would register a a control-left-click.

**Comment by lytex** [comment]
It's important to turn off the sticky key after the next event happens, for example sticky Shift is very useful to me since it avoids "DOuble capitalization" if I happen to press the `o` key before releasing Shift. Maybe for Control is not as critical but still I want to retain that behavior to avoid sending Control twice.

I wouldn't mind however having to wait a little to use Ctrl+Left Click, I think a good solution would be sending Ctrl if the key is pressed more than, say 1s, then Ctrl is pressed for another 1s

Any idea on how to do that? I'm trying to combine `tap-*` functions with `sticky-keys` but it feels a bit hacky

**Comment by david-janssen** [comment]
That is a very pretty solution :-).

I've decided I would like to add mouse-event handling to KMonad, even if only clicks (and not motions). However, this would be a new feature, and I want to first fix a bunch of bugs and refactor certain modules. So, low on the list, but maybe someday. Can I close the issue since the practical how-do-I-get-the-behavior-I-want issue is basically solved?

**Comment by lytex** [comment]
Yes, the solution mostly works. There are a couple of minor issues, if I solve them I'll post the solution here:

1. Reducing the timeout of `tap-hold` improves the snappiness of some shortcuts (Alt+1, Alt+2 etc) so that for example `(tap-hold 50 (around-next-timeout 500 lctl XX) lctl)` is a better solution. WIth a timeout of 1000ms, sometimes I  
2. When pressing combos such as Ctrl+Alt+V one has to release first the non-modifier key (the V in this case), otherwise Ctrl+V or Alt+V is sent instead, depending on the key release order.
